### PR TITLE
Update colors

### PR DESCRIPTION
**Pull request by im-adithya**
## Description
This PR updates the BRAND_COLOR and HIGHLIGHT_COLOR to match the new logo.

### CODE REVIEW

**Comment by WesleyAC** [comment]
Could you add a comment with a link back to the file that has these colors in the webapp, so it's easier to find where they come from, in case we need to update them in the future?

Also, should the "Connecting" spinner have the new color? It's hard to see what it looks like in the gif, but it seems like it might still be the old color.

**Comment by chrisbobbe** [comment]
> Could you add a comment with a link back to the file that has these colors in the webapp, so it's easier to find where they come from, in case we need to update them in the future?

I'd [recommended](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/design.20issues/near/1130309) trying out `rgb(100, 146, 253.5)` based on Anders's [report](https://github.com/zulip/zulip-mobile/pull/4467#issue-567898180) that it's the midpoint of the logo's gradient, which seemed like as good a place to start as any.

But even better would be if there's a file we can point to in the webapp; that'd be easier to sync with over time, when any changes get made. @andersk, do you know of such a file?

File: src/styles/constants.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ export const CONTROL_SIZE = 44;
 export const NAVBAR_SIZE = 58;
 
 /* This color is also used directly in several places in our WebView's CSS. */
-export const BRAND_COLOR = 'hsl(170, 48%, 54%)';
+export const BRAND_COLOR = 'hsl(222, 97%, 69%)';
 export const BORDER_COLOR = BRAND_COLOR;
-export const HIGHLIGHT_COLOR = 'hsla(187, 35%, 51%, 0.5)';
+export const HIGHLIGHT_COLOR = 'hsla(222, 97%, 69%, 0.5)';
```
Comment (chrisbobbe) [inline src/styles/constants.js:9]: Could you please remind me why we're changing `HIGHLIGHT_COLOR`? In particular, I notice the following:

- Before this commit, `HIGHLIGHT_COLOR` has different HSL values than `BRAND_COLOR`.
- After this commit, `HIGHLIGHT_COLOR` has the same HSL values as `BRAND_COLOR`.

What's going on there? 🙂 If you dig through the [history](https://github.com/zulip/zulip-mobile/blob/master/docs/howto/git.md#reading-history) of the file, can you find any clues about why they might have been different, before?

**Review by chrisbobbe — COMMENTED** [review]
Thanks, @im-adithya! See some comments below.

Also, linking some prior conversation on CZO: https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/design.20issues/near/1130309

**Comment by im-adithya** [comment]
> Could you add a comment with a link back to the file that has these colors in the webapp, so it's easier to find where they come from, in case we need to update them in the future?

I think there's no such file that states the color changes as far as I remember...

> Also, should the "Connecting" spinner have the new color? It's hard to see what it looks like in the gif, but it seems like it might still be the old color.

Sure! On it

File: src/styles/constants.js (hunk 1 above)
Comment (im-adithya) [reply src/styles/constants.js:9]: Sorry @chrisbobbe, I didn't notice the fact that they were different previously, will browse through and tell you if there is any reason behind it!

File: src/styles/constants.js [hunk 2]
```diff
@@ -4,8 +4,8 @@ export const CONTROL_SIZE = 44;
 export const NAVBAR_SIZE = 58;
 
 /* This color is also used directly in several places in our WebView's CSS. */
```
[the remark this answers is not in the document]
Comment (im-adithya) [reply src/styles/constants.js:6]: Sure!

File: src/styles/constants.js (hunk 1 above)
Comment (im-adithya) [reply src/styles/constants.js:9]: After digging the history, I've come to know that this was just an arbitrary value used while reworking on the common styling by Boris in 2017
https://github.com/zulip/zulip-mobile/commit/08c0cbd287904969a2361baa8c8c959e247b80ba#diff-395692415cbb03acdc7f24b1058a24796b7f881c6918ad7a9679cd61f4954863R11

It was then shifted from styles/index.js to the current styles/constants.js file
https://github.com/zulip/zulip-mobile/commit/a4c2c1a6586239d6cc3f2987ba9b29fac57a7570#diff-a81e34235c2a8da54ccf35b5c75a38c728b0c4915f5497604578220b120f9915R8

And during another commit, it was decided to use hsl values instead of rgb ones...
https://github.com/zulip/zulip-mobile/commit/72c0868704aaf383e54cab3de33416bad04f597b#diff-a81e34235c2a8da54ccf35b5c75a38c728b0c4915f5497604578220b120f9915R8

...which introduced decimal values rounded later in this commit giving us the present values for HIGHLIGHT_COLOR (and BRAND_COLOR as well)
https://github.com/zulip/zulip-mobile/commit/4d075a560ca00df95977c5e7d18583857eb5d3f1#diff-a81e34235c2a8da54ccf35b5c75a38c728b0c4915f5497604578220b120f9915R8


So, what should we do now? @chrisbobbe 
Can we stick to using the BRAND_COLOR by decreasing its alpha value? Or should we take a random value again?

Comment (chrisbobbe) [reply src/styles/constants.js:9]: Great, thanks for finding all that history! I agree that there doesn't seem to be an explanation. Ah, well.

Looking at where HIGHLIGHT_COLOR is used (just in src/common/Touchable.js for when a `Touchable` during the time it's pressed/selected), I'd be very happy with your solution of making it the same as BRAND_COLOR with a different alpha value. It's simpler, and if there's any reason to prefer the changed HSL values based on color theory or good design practices, such a reason is not obvious to me, and it's not documented. 🙂

I think a good, clear way to make this change (such that our work is more helpful to future developers than what we've just found looking in the history so far! 🙂) would be something like this, starting from the beginning of the branch, before the more interesting changes in this PR:

- A commit that changes HIGHLIGHT_COLOR to `hsl(170, 48%, 54%, 0.5)`, i.e., the same as BRAND_COLOR but with the different alpha value. The commit message references and/or summarizes this conversation for why we'd like to make that change.
- Maybe an [`[nfc]` commit](https://github.com/zulip/zulip-mobile/blob/master/docs/glossary.md#nfc-or-nfc) that makes the connection with BRAND_COLOR even more explicit. For example, like this:

  ```js
  import Color from 'color';
  
  // ...
  
  export const HIGHLIGHT_COLOR = Color(BRAND_COLOR)
    .fade(0.5)
    .toString();
  ```
  
  (See the docs of the NPM package `color` that we use: https://www.npmjs.com/package/color#manipulation.)
- Then, the PR proceeds with the color-change from green to blue—except this time HIGHLIGHT_COLOR doesn't have to change; it'll already be computed to be the value we want. 🙂

**Comment by chrisbobbe** [comment]
> I think there's no such file that states the color changes as far as I remember...

Ah, well. So, for the critical step of explaining how we chose the new color, we can just reference Anders' [report](https://github.com/zulip/zulip-mobile/pull/4467#issue-567898180) that this color is the midpoint of the logo's gradient.

File: src/styles/constants.js (hunk 1 above)
Comment (im-adithya) [reply src/styles/constants.js:9]: Nice! I will make the necessary changes!

**Comment by im-adithya** [comment]
@chrisbobbe This is ready for review :)

File: src/webview/static/base.css [hunk 3]
```diff
@@ -402,9 +402,9 @@ body {
   vertical-align: top;
 }
 .self-voted {
-  color: hsl(170, 48%, 54%);
-  border: 1px solid hsl(170, 48%, 54%);
-  background: hsla(177.1, 69.7%, 46.7%, 0.1);
+  color: hsl(222, 99%, 69%);
+  border: 1px solid hsl(222, 99%, 69%);
+  background: hsla(222, 99%, 69%, 0.1);
```
Comment (chrisbobbe) [inline src/webview/static/base.css:407]:
```
[...] and uses the
(new) BRAND_COLOR with a reduced alpha value for the background of
'.self-voted' class which used to be an arbitrary value previously.
```

This is an interesting change that should go into its own commit.

**Review by chrisbobbe — CHANGES_REQUESTED** [review]
Thanks, @im-adithya! See a few comments below.

As this is getting closer, I've added the commit to update an iOS file, as promised, and pushed it to this branch (along with a rebase atop the current `master`). 🙂

File: src/common/ZulipSwitch.js [hunk 4]
```diff
@@ -24,14 +25,16 @@ export default class ZulipSwitch extends PureComponent<Props> {
...
       <Switch
         value={value}
         trackColor={{
           false: 'hsl(0, 0%, 86%)',
-          true: BRAND_COLOR,
+          true: Color(BRAND_COLOR)
+            .fade(0.7)
+            .toString(),
         }}
+        thumbColor={value ? BRAND_COLOR : '#f1f1f1'}
```
[the remark this answers is not in the document]
Comment (im-adithya) [reply src/common/ZulipSwitch.js:37]: > Ah, but also: Where does #f1f1f1 for the "off" position come from? 🙂

It is a bad fix, but I had no other option as 'unset' or 'none' aren't working at all for the thumbColor prop.
What I mean by that is:
Screenshot from 2021-04-06 23-37-09
The thumbColor sets to the available value (even when it isn't true) whenever an invalid value is set. This isn't an issue with the boolean `value` as it appears to be working whenever we set a valid color for the false condition (just like now, #f1f1f1 is applied when the `value` equals false. And I didn't find any default option for this prop on the internet as well.

I also tried using null and 0, but that lead to removing the thumb as a whole :slightly_smiling_face:  (whenever it is switched off)

File: src/styles/constants.js [hunk 5]
```diff
@@ -1,11 +1,14 @@
 /* @flow strict-local */
+import Color from 'color';
 
 export const CONTROL_SIZE = 44;
 export const NAVBAR_SIZE = 58;
 
 /* This color is also used directly in several places in our WebView's CSS. */
-export const BRAND_COLOR = 'hsl(170, 48%, 54%)';
+export const BRAND_COLOR = 'hsl(222, 99%, 69%)';
```
[the remark this answers is not in the document]
Comment (im-adithya) [reply src/styles/constants.js:8]: Sure @chrisbobbe!

File: src/common/ZulipSwitch.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (im-adithya) [reply src/common/ZulipSwitch.js:37]: > (I didn't notice any of this in my earlier reviews, since at that point I'd done most of my manual testing on iOS.)

I'm curious about how this looks on iOS now.

[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/common/ZulipSwitch.js:37]: Ah, OK, got it.

I think Material Design would have us using its `colorSurface` value. See "Color" in the table under "Thumb attributes" in [this Material Design doc](https://material.io/components/switches/android#anatomy-and-key-properties). (Huh, and it would have us using a secondary color, instead of the primary color, for the "checked" state. I guess we can worry about that later, if at all.)

But there's quite a bit of [setup](https://material.io/develop/android/docs/getting-started) involved if we want to go farther down the paths Material Design would have us take. In particular, I think that work would be required if we want to hook into a `colorSurface` value the right way.

Checking to see if we have a variable that we use roughly the same way as `colorSurface`, I think it might be `backgroundColor`, from the ThemeContext. Here's what a [Material Design doc](https://material.io/develop/android/theming/color) says about `colorSurface`: "Surface colors affect surfaces of components, such as cards, sheets, and menus." Not too descriptive, I guess. I believe (from [this doc](https://material.io/develop/android/theming/dark)) that `colorSurface` varies between light and dark modes, and our `backgroundColor` does that too.

OK, here's what it looks like with `backgroundColor`, since I was curious and tried it out. In light and dark modes, on Android:



Hmm. I'm actually not so sure I like how it looks in dark mode; the thumb looks too dark, to me. But I'm not a regular Android user. Would you see that and think it looks like what other apps do?

@gnprice, should I open an issue for following [this setup](https://material.io/develop/android/docs/getting-started) to allow a deeper connection to Material Design color patterns on Android? I'm even less familiar with a comparable setup for iOS, but I think the place to start there would be [this doc](https://material.io/develop/ios).

**Comment by im-adithya** [comment]
@chrisbobbe I have made the changes.
But why did that remove your commit? (I'm sorry for that I didn't know)

**Comment by chrisbobbe** [comment]
> @chrisbobbe I have made the changes.
> But why did that remove your commit? (I'm sorry for that I didn't know)

Ah, no problem! 🙂 I can easily add it back. This has happened to me before, too.

As you know, when you force-push a branch to GitHub, it overwrites the branch on the remote with the new commits; it doesn't try to do anything with the commits in the PR's previous revision. This is good, because that's usually what we want to do along the lifetime of a PR; we continually improve the commits on the remote by overwriting them, until the PR is ready to be merged. I did a force-push to the `task-change-color` on your remote, `git@github.com:im-adithya/zulip-mobile`.

So, when you made changes and did a force-push, naturally it forgot about the revision that I had force-pushed.

The way around this is to save the changes from the remote branch, to your local branch. If `me` is the alias you use for your remote, this could look like `git pull --ff-only me task-change-color`. Then, you'd be working with the revision I had pushed. 🙂

File: src/common/ZulipSwitch.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (im-adithya) [reply src/common/ZulipSwitch.js:37]: >  the thumb looks too dark, to me.

Yes, it is definitely looking odd.

> Would you see that and think it looks like what other apps do?

Seems like everyone is using a light color for the thumb...

Discord:
<p float="left">
  
  
</p>
Slack:
<p float="left">
  
  
</p>
Rocket.Chat:
<p float="left">
  
  
</p>

**Comment by im-adithya** [comment]
Got it! Thank you @chrisbobbe 🙌

File: src/common/ZulipSwitch.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/common/ZulipSwitch.js:37]: Thanks for grabbing all those screenshots; it's really helpful to see what other apps do!

Yes, I agree; I think a light "thumb" looks better, even in dark mode. Could go with `#FFFFFF`, since that's `colorSurface`'s default value in light mode. Then have a code comment mentioning `colorSurface`, in case we get around to using such a thing. For example, something like

```js
        thumbColor={
          /* eslint-disable operator-linebreak */
          value
            ? // Material design would actually have this be a secondary
              // color, not a primary color. See "Thumb attributes" at
              // this doc:
              //   https://material.io/components/switches/android#anatomy-and-key-properties
              BRAND_COLOR
            : // Material design would have this be `colorSurface`
              // (see above-linked doc), which defaults to `#FFFFFF`,
              // at least in light mode; see
              //   https://material.io/develop/android/theming/color.
              '#FFFFFF'
        }
```

[the remark this answers is not in the document]
Comment (im-adithya) [reply src/common/ZulipSwitch.js]: Done!

**Comment by gnprice** [comment]
Thanks @im-adithya and @chrisbobbe ! Sorry for the slow review -- this looks good, and I've just merged.

I spent a bit of time playing around with this on my Android phone, and the one thing that looked off is that it was hard to see the switches when they were on, at least in dark mode. So I added a small commit to increase the alpha on that, as well as one just on some comments:

91ad1a7b2 styles: Make switch's track less transparent.
70bed8dd1 styles [nfc]: Tweak comment on BRAND_COLOR.

File: src/styles/constants.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (gnprice) [reply src/styles/constants.js:11]: (FTR I think the ideal place for that URL is in the commit message -- in my added commit before merge I made the comment look like this:
```
// The value `hsl(222, 99%, 69%)` is chosen to match `rgb(100, 146, 253.5)`,
// which is the sRGB midpoint of the Zulip logo's gradient.
//
// Note this color is also used directly in several places in our WebView's CSS.
export const BRAND_COLOR = 'hsl(222, 99%, 69%)';
```
)

[the remark this answers is not in the document]
Comment (chrisbobbe) [reply src/styles/constants.js:11]: Noted, thanks!
